app/javascript/spree/dashboard/controllers/product_edit_controller.js

```diff
@@ -0,0 +1,41 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["availableOn", "makeActiveAt", "discontinueOn", "status"]
+
+  initialize() {
+    $(this.statusTarget).on("select2:select", function (e) {
+      let event = new Event("change")
+      e.target.dispatchEvent(event)
+    })
+  }
+
+  connect() {
+    this.statusTarget.dispatchEvent(new Event("change"))
+  }
+
+  switchAvailabilityDatesFields(event) {
+    let status = event.target.value
+    if (status === "draft") {
+      this.show(this.availableOnTarget)
+      this.show(this.makeActiveAtTarget)
+      this.hide(this.discontinueOnTarget)
+    } else if (status === "active") {
+      this.hide(this.availableOnTarget)
```

Comment:
@kgorazd available on should still be visible, this is the release date

Reply:
done